common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -132,6 +134,47 @@ def test_staff_get_all_entitlements(self):
         results = response.data.get('results', [])
         assert results == CourseEntitlementSerializer(entitlements, many=True).data
 
+        # Make sure that these results aren't expired
+        assert results[0].get('expired_at') is None and results[1].get('expired_at') is None
+
+    def test_staff_get_expired_entitlements(self):
+        past_datetime = datetime.utcnow().replace(tzinfo=pytz.UTC) - timedelta(days=365 * 2)
+        entitlements = CourseEntitlementFactory.create_batch(2, created=past_datetime)
+
+        # Set the first entitlement to be at a time that it isn't expired
+        entitlements[0].created = datetime.utcnow()
+        entitlements[0].save()
+
+        response = self.client.get(
+            self.entitlements_list_url,
+            content_type='application/json',
+        )
+        assert response.status_code == 200
+        results = response.data.get('results', [])  # pylint: disable=no-member
+        # Make sure that the first result isn't expired, and the second one is also not for staff users
+        assert results[0].get('expired_at') is None and results[1].get('expired_at') is None
```

Comment:
It might be good to assert that `results[1].expired_at_datetime` is not `None` to show that it otherwise would have been expired if we didn't add the special logic to the view.